### PR TITLE
bind docker container to internal port if using vhosts

### LINKED ISSUE #860: Bind nginx requests directly to docker internal IP
This would help avoid cases where users may not being using a firewall to block random ports *and* they are using VHOST-based deploys. Apps should only be routed to via the private docker ip address, not a public docker port.

Refs #207.
Refs #557.

@michaelshobbs I think you might be more familiar with how this works, so if you have time, would you mind taking a look at those PRs and modifying it to work in the new setup?

### PR DESCRIPTION
**Pull request by michaelshobbs**
Per IRC conversations, this was implemented as a (very simple) pluginhook. Includes tests.

close #860 
refs #207
refs #557

### CODE REVIEW

File: plugins/nginx-vhosts/bind-external-ip [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+set +e; NO_VHOST=$(dokku config:get $APP NO_VHOST); set -e
+if [[ -f "$DOKKU_ROOT/VHOST" ]];then
+  GLOBAL_VHOST=$(< "$DOKKU_ROOT/VHOST")
+else
+  GLOBAL_VHOST=$(< "$DOKKU_ROOT/HOSTNAME")
+fi
+
+if [[ -n "$NO_VHOST" ]]; then
+  echo true # bind to external ip. VHOST is disabled for this app
+elif [[ "$GLOBAL_VHOST" =~ ([0-9]{1,3}[\.]){3}[0-9]{1,3} ]];then
```
Comment (josegonzalez) [inline plugins/nginx-vhosts/bind-external-ip]: IPv6?

**Comment by michaelshobbs** [comment]
Crap! I did not take that into account in this or the previous implementation. I'll take a look in the morning.

**Comment by michaelshobbs** [comment]
Snagged an IPv6 regex from SO (http://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses) and put it into the plugin and the domains one as well.

**Comment by josegonzalez** [comment]
Build failed. Womp.

**Comment by michaelshobbs** [comment]
Docker is having issues according to their rss

**Comment by josegonzalez** [comment]
Can you add some docs to this and then we can merge?

**Comment by michaelshobbs** [comment]
Sure thing. Preference on the best place it should live? New md file or existing?

**Comment by michaelshobbs** [comment]
Probably belongs in nginx, no?

**Comment by josegonzalez** [comment]
In nginx probably. Feel free to reorder stuff as necessary.

**Comment by josegonzalez** [comment]
Nice work. Merging and shipping a new release :)

**Pull request merged**